File: client/services/cricket.ts
import apiClient from "@/lib/apiClient";

export async function fetchLiveCricketMatches() {
  const response = await apiClient.get("/cricket/live");
  return response.data;
}

export async function fetchCricketMatchInfo(matchId: string) {
  const response = await apiClient.get(`/cricket/match/${matchId}`);
  return response.data;
}

export async function fetchCricketSeries() {
  const response = await apiClient.get("/cricket/series");
  return response.data;
}

File: server/app/api/routers/cricket.py
from fastapi import APIRouter
from app.services.cricket_service import (
    get_live_cricket_matches,
    get_cricket_match_info,
    get_cricket_series,
)

router = APIRouter(prefix="/cricket", tags=["cricket"])


@router.get("/live")
async def live_cricket():
    return await get_live_cricket_matches()


@router.get("/match/{match_id}")
async def cricket_match(match_id: str):
    return await get_cricket_match_info(match_id)


@router.get("/series")
async def cricket_series():
    return await get_cricket_series()